src/joy_api.c

```diff
@@ -109,75 +109,137 @@ static unsigned int joy_splt_format_data(flow_record_t *rec,
                                          unsigned char *data)
 {
     unsigned int i = 0;
+    unsigned int entry_counter = 0;
     unsigned int entries_used = 0;
     unsigned int num_of_pkts = 0;
     unsigned int data_len = 0;
+    uint16_t last_len = 0;
+    int16_t run_count = 0;
     struct timeval ts;
     uint16_t *formatted_data = (uint16_t*)data;
 
-    /* see how many packets we have to process - max is MAX_NFV9_SPLT_SALT_PKTS */
-    num_of_pkts = (rec->op < MAX_NFV9_SPLT_SALT_PKTS) ? rec->op : MAX_NFV9_SPLT_SALT_PKTS;
+    /* figure out actual entries needed
+     * we max out at 10 entries for a SPLT record.
+     */
+    for (i=0; i < rec->op; ++i) {
+        if (rec->pkt_len[i] != last_len ) {
+            /* run has ended increment entry count */
+            if (run_count < 0) {
+                ++entry_counter;
+            }
+
+            /* store last length */
+            last_len = rec->pkt_len[i];
+
+            /* increment entry count, reset run */
+            ++entry_counter;
+            run_count = 0;
+
+            /* stop if we reach the entry limit */
+            if (entry_counter == MAX_NFV9_SPLT_SALT_PKTS) {
+                break;
+            }
+        } else {
+            /* found a run, start the count */
+            --run_count;
+
+            /* stop if we reach the entry limit */
+            if (entry_counter == (MAX_NFV9_SPLT_SALT_PKTS-1)) {
```

Comment:
Is the entry limit 1 less than MAX_NFV9_SPLT_SALT_PKTS? Different entry limit on line 139

Reply:
yes it is different because of the run. We don't want to end the list on a run of values.

Reply:
Suggest commenting it differently than the line above. Both say the same thing. We should state that we don't want to end on run of values.

Reply:
its in the comment right below where the entry count is adjusted.

    /* if we had an active run, increment entry count
     * this is why the run count leg break 1 entry early
     */

Reply:
Yeah, I was looking at this one /* stop if we reach the entry limit */